src/ucp/proto/proto_select.h

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCP_PROTO_SELECT_H_
+#define UCP_PROTO_SELECT_H_
+
+#include "proto.h"
+
+#include <ucs/datastruct/khash.h>
+
+
+/**
+ * Some flags from ucp_request_param_t.op_attr_mask can affect protocol
+ * selection decision.
+ */
+#define UCP_PROTO_SELECT_OP_ATTR_BASE   UCP_OP_ATTR_FLAG_NO_IMM_CMPL
```

Comment:
why need to redefine, but not use original consts?

Reply:
this is a different usage: the purpose is to define that the first flag from op_attr_mask to use is UCP_OP_ATTR_FLAG_NO_IMM_CMPL (it's used to put only the relevant flags to the select key)